pybaseball/statcast_pitcher_spin.py

```diff
@@ -0,0 +1,140 @@
+from pybaseball import statcast_pitcher
+import pandas as pd
+import numpy as np
+
+K = .005153 #Environmental Constant
+SIG_DIG = 4
+
+def statcast_pitcher_spin(start_dt=None, end_dt=None, player_id=None):
+	pitcher_data = statcast_pitcher(start_dt, end_dt, player_id)
+
+	spin_df = pitcher_data[['release_extension', 'vx0', 'vy0', 'vz0', 'ax', 'ay', 'az', 'release_spin_rate']].copy()
+
+	spin_df = find_release_point(spin_df)
+	spin_df = find_release_time(spin_df)
+	spin_df = find_release_velocity_components(spin_df)
+	spin_df = find_flight_time(spin_df)
+	spin_df = find_average_velocity_components(spin_df)
+	spin_df = find_average_velocity(spin_df)
+	spin_df = find_average_drag(spin_df)
+	spin_df = find_magnus_acceleration_magnitude(spin_df)
+	spin_df = find_average_magnus_acceleration(spin_df)
+	spin_df = find_magnus_magnitude(spin_df)
+	spin_df = find_phi(spin_df)
+	spin_df = find_lift_coefficient(spin_df)
+	spin_df = find_spin_factor(spin_df)
+	spin_df = find_transverse_spin(spin_df)
+	spin_df = find_spin_efficiency(spin_df)
+	spin_df = find_theta(spin_df)
+
+	pitcher_data[['Mx', 'Mz', 'phi', 'theta']] = spin_df[['Mx', 'Mz', 'phi', 'theta']].copy()
+	return pitcher_data
+
+# def get_statcast_pither_test_data():
+# 	df = pd.read_csv("tests/statcast_pitching_test_data.csv")
+# 	return df
+
+def find_release_point(df):
+	df['yR'] = (60.5 - df['release_extension']).round(SIG_DIG)
+	return df
+
+def find_release_time(df):
+	df['tR'] = time_duration(df['yR'], df['vy0'], df['ay'], 50, False).round(SIG_DIG)
```

Comment:
should 50 be variable at top of the file?

Reply:
Both of these numbers are actually explained in the README tab of Prof. Nathan's excel workbook. Y Velocity is captured at Y=50 and the final data points are taken at y=17/12. These numbers will be defined at the top of the file as such.